rapidfuzz/fuzz.impl

```diff
@@ -160,15 +160,15 @@ partial_ratio_long_needle(InputIt1 first1, InputIt1 last1, InputIt2 first2, Inpu
     for (const auto& block : blocks) {
         if (block.length == len1) {
             res.score = 100;
-            res.dest_start = std::max(int64_t(0), block.dpos - block.spos);
+            res.dest_start = std::max(static_cast<std::ptrdiff_t>(0), block.dpos - block.spos);
```

Comment:
I think here I would prefer:
```
res.dest_start = std::max<std::ptrdiff_t>(0, block.dpos - block.spos);
```
simply because C++ casts look horrendous

Reply:
> simply because C++ casts look horrendous

haha, I agree. There are some other usages of casts within `std::max` or `std::min`. They are found in `rapidfuzz/details/types.hpp`. For example, in line 159:
```cpp
start = std::max((int)(start + (int)vec.size()), 0);
```
Do you think these need to conform to the new style as well?

Reply:
Yes it would probably make sense to get rid of c style casts in most places. So in this example:
```cpp
start = std::max<int>(start + static_cast<int>(vec.size()), 0);
```
In addition it would probably make sense to replace teh usage of int in this function with `std::ptrdiff_t`.